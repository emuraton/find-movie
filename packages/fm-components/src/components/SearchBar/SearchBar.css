.SearchBar {
  width: 100%;
  background: #ffffff;
  box-sizing: border-box;
  border-bottom: 1px solid #d8d8d8;
}

.SearchBar-form {
  display: flex;
  justify-content: space-between;
  width: 100%;
  height: 44px;
}

.SearchBar-iconParent {
  height: 100%;
  width: 40px;
  order: 1;
  margin-right: 30px;
}

.SearchBar-button {
  height: 100%;
  margin: 0 0 0 auto;
  display: block;
}

.SearchBar-icon {
  max-height: 100%;
  padding: 12px;
  box-sizing: border-box;
}

.SearchBar-input {
  flex: 1;
  background-color: transparent;
  color: #232323;
  border: none;
  outline: none;
  box-sizing: border-box;
}
