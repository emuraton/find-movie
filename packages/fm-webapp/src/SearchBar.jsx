import React from 'react'
import PropTypes from 'prop-types'


export default class SearchBar extends Components {
  constructor(props) {
    super(props)
  }

  render() {
    return (
      <form className="SearchBar-form" onSubmit={this.onSubmit}>
        <button className=""></button>
        <input>

        </input>
      </form>
    )
  }
}

SearchBar.PropTypes = {
  onSubmit: PropTypes.func.isRequired
}
