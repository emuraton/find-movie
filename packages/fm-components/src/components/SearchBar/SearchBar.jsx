import React, { Component } from 'react'
import PropTypes from 'prop-types'

import searchImg from './images/search-icon.svg'
require('./SearchBar.css')

export default class SearchBar extends Component {
  constructor(props) {
    super(props)
  }

  render() {
    const { onSubmit } = this.props
    return (
      <div className="SearchBar">
        <form className="SearchBar-form" onSubmit={onSubmit}>
          <div className="SearchBar-iconParent">
            <button className="SearchBar-button" onClick={onSubmit}>
              <img className="SearchBar-icon" alt="Search" src={searchImg} />
            </button>
          </div>
          <input className="SearchBar-input" type="text" placeholder="Search for a movie, tv show, person..."/>
        </form>
      </div>
    )
  }
}

SearchBar.PropTypes = {
  onSubmit: PropTypes.func.isRequired
}
