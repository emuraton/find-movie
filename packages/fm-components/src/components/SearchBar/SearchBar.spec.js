import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import SearchBar from './SearchBar'

describe('<SearchBar />', () => {

  const initialProps = {
    onSubmit: jest.fn()
  }

  describe('@renders', () => {
    it('in default state', () => {
      const wrapper = shallow(<SearchBar {...initialProps} />)
      expect(toJson(wrapper))
        .toMatchSnapshot()
    })
  })

  describe('@events', () => {
    beforeEach(() => jest.resetAllMocks())

    describe('onSubmit', () => {
      it('should call onSubmit()', () => {
        const wrapper = shallow(<SearchBar {...initialProps} />)
        expect(initialProps.onSubmit).not.toBeCalled()
        wrapper.find('.SearchBar-form').simulate('submit', { preventDefault: jest.fn()})
        expect(initialProps.onSubmit).toHaveBeenCalledTimes(1)
      })

      it('should all onSubmit() via onClick event', () => {
        const wrapper = shallow(<SearchBar { ...initialProps} />)
        expect(initialProps.onSubmit).not.toBeCalled()
        wrapper.find('.SearchBar-button').simulate('click', { preventDefault: jest.fn()})
        expect(initialProps.onSubmit).toHaveBeenCalledTimes(1)
      })
    })

    describe('onChange', () => {
      const testQuery = 'testQuery'
      it('should save event value', () => {
        const wrapper = shallow(<SearchBar {...initialProps} />)
        const instance = wrapper.instance()
        expect(instance.state).toMatchSnapshot()
        wrapper.find('.SearchBar-input').simulate('change', { target: { name: 'name', value: testQuery }})
        expect(instance.state).toMatchSnapshot()
      })
    })
  })
})
