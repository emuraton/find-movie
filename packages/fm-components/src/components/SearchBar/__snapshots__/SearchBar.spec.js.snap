// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SearchBar /> @renders in default state 1`] = `
<div
  className="SearchBar"
>
  <form
    className="SearchBar-form"
    onSubmit={[Function]}
  >
    <div
      className="SearchBar-iconParent"
    >
      <button
        className="SearchBar-button"
        onClick={[Function]}
      >
        <img
          alt="Search"
          className="SearchBar-icon"
          src={
            Object {
              "process": [Function],
            }
          }
        />
      </button>
    </div>
    <input
      className="SearchBar-input"
      placeholder="Search for a movie, tv show, person..."
      type="text"
    />
  </form>
</div>
`;
